inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1831,112 +1831,112 @@ void FuseFQIntoWeightsPass::run() {
+                        << " Per-channel quantization ";
+                }
 
-            if (inputRange.first.size() != outputRange.first.size() ||
-                inputRange.second.size() != outputRange.second.size()) {
-                THROW_GNA_LAYER_EXCEPTION(fqLayer) << " size of input and output range differs. "
-                    << "input min size: " << inputRange.first.size() << ", "
-                    << "output min size: " << outputRange.first.size() << ", "
-                    << "input max size: " << inputRange.second.size() << ", "
-                    << "output max size: " << outputRange.second.size();
+                // check if
+                // - weights were float values and need to be quantized,
+                // - weights are integer values and quantization can be skipped
+                quantized->_weights_quant.SetMinValues(inputRange.first, true);
+                quantized->_weights_quant.SetMaxValues(inputRange.second, true);
+                quantized->_weights_quant.SetMinValues(outputRange.first, false);
+                quantized->_weights_quant.SetMaxValues(outputRange.second, false);
+                quantized->_weights_quant.SetLevels(levels);
+
+                // lets find out minimum scale factor among channels
+                if (!quantized->_weights_quant.IsStatsSet()) {
+                    THROW_GNA_LAYER_EXCEPTION(fqLayer) << " per channel/tensor weigths scales are missed";
+                }
+                continue;
             }
 
-            if (levels > std::numeric_limits<uint8_t>::max() && outputRange.first.size() > 1) {
-                THROW_GNA_LAYER_EXCEPTION(fqLayer) << " unsupported per-channel quantization for int16 weights."
-                    << " Per-channel quantization ";
-            }
+            size_t depth = 1;
+            intel_dnn_component_t component;
+            component.num_columns_in = weightDims[1];
+            component.num_rows_in = weightDims[0];
 
-            // check if
-            // - weights were float values and need to be quantized,
-            // - weights are integer values and quantization can be skipped
-            quantized->_weights_quant.SetMinValues(inputRange.first, true);
-            quantized->_weights_quant.SetMaxValues(inputRange.second, true);
-            quantized->_weights_quant.SetMinValues(outputRange.first, false);
-            quantized->_weights_quant.SetMaxValues(outputRange.second, false);
-            quantized->_weights_quant.SetLevels(levels);
-
-            // lets find out minimum scale factor among channels
-            if (!quantized->_weights_quant.IsStatsSet()) {
-                THROW_GNA_LAYER_EXCEPTION(fqLayer) << " per channel/tensor weigths scales are missed";
+            if (LayerInfo(weightableLayer).isConvolution()) {
+                depth = (weightDims.size() == 4) ? weightDims[3] : 1;
```

Comment:
should we take into account weightDims[2] for convolution since sometimes it's not equal to 1 (even for 1d convolution)?